### PR TITLE
Fixed #458 - Upgrade Jackson to v2.5.0

### LINKED ISSUE #458: Upgrade Jackson to latest (2.5.0)
Current version of CBL Android uses Jackson 1.9.2. 
Latest version of Jackson is 2.5.0. (http://jackson.codehaus.org/)

### PR DESCRIPTION
**Pull request by hideki**

### CODE REVIEW

**Comment by tleyden** [comment]
@hideki can you try to build on the jenkins machine and make sure it compiles there and there are no new unit test failures?

**Comment by pasin** [comment]
@tleyden Can you help reviewing this PR? I don't know the history of using Jackson with SQLite blob (byte arrays with zero terminated).

**Comment by hideki** [comment]
http://172.23.113.52:8080/job/couchbase-lite-android-build/46/

**Comment by hideki** [comment]
It seems that Jenkins build and unit tests passed.

File: src/main/java/com/couchbase/lite/support/JsonDocument.java [hunk 1]
```diff
@@ -36,7 +36,13 @@ public Object jsonObject() {
                 tmp = new LazyJsonArray<Object>(json);
             } else {
                 try {
-                    tmp = Manager.getObjectMapper().readValue(json, Object.class);
+                    // this is for Jackson 2.5.0
```
Comment (tleyden) [inline src/main/java/com/couchbase/lite/support/JsonDocument.java]: I'm confused about what this is doing here.. why is this needed?

Comment (hideki) [inline src/main/java/com/couchbase/lite/support/JsonDocument.java]: json is byte[] which is from `Cursor.getBlob()`. And json byte array is ended with '\0'. '\0' causes parsing problem with Jackson 2.5.0. We did not observe this problem with Jackson 1.9.2.

Comment (tleyden) [inline src/main/java/com/couchbase/lite/support/JsonDocument.java]: ok that makes sense.  can you add that to the comments?

so does that mean we are giving invalid json to Jackson?

Comment (hideki) [inline src/main/java/com/couchbase/lite/support/JsonDocument.java]: - Added comment, and pushed it.
- As byte[] has length property, it does not require NULL stop like C char array. So It seems Jackson 2.5.0 treats '\0' as part of data, but it is not able to handle it. I believe `Cursor.getBlob()` does not need to added '\0' at end of byte[], also Jackson 2.5.0 should handle it.
